src/main/java/org/jabref/preferences/JabRefPreferences.java

```diff
@@ -2014,18 +2016,30 @@ public InternalPreferences getInternalPreferences() {
         return internalPreferences;
     }
 
-    private String getUser() {
-        if (StringUtil.isNotBlank(userName)) {
-            return userName;
+    String getUserAndHost() {
+        if (StringUtil.isNotBlank(userAndHost)) {
+            return userAndHost;
         }
 
-        try {
-            userName = get(DEFAULT_OWNER) + '-' + InetAddress.getLocalHost().getHostName();
-            return userName;
-        } catch (UnknownHostException ex) {
-            LOGGER.error("Hostname not found. Please go to https://docs.jabref.org/ to find possible problem resolution", ex);
-            return get(DEFAULT_OWNER);
+        String hostName;
+        // Following code inspired by https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/SystemUtils.html#getHostName--
+        // See also https://stackoverflow.com/a/20793241/873282
+        if (OS.WINDOWS) {
+            hostName = System.getenv("COMPUTERNAME");
+        } else {
+            hostName = System.getenv("HOSTNAME");
         }
+        if (StringUtil.isBlank(hostName)) {
+            try {
+                hostName = InetAddress.getLocalHost().getHostName();
+            } catch (UnknownHostException e) {
+                LOGGER.error("Hostname not found. Please go to https://docs.jabref.org/ to find possible problem resolution", e);
```

Comment:
Make this a warning and say that the hostname will be set to localhost.

Reply:
By chance, do you know whether the documentation has info about that? 😅

Reply:
Of course not 😆